### PR TITLE
Allow to sort products by sku

### PR DESCRIPTION
**Pull request by szymoniwacz**

### CODE REVIEW

File: core/app/sorters/spree/products/sort.rb [hunk 1]
```diff
@@ -30,6 +31,18 @@ def by_price(scope)
...
+      def sku?
+        sort_field == 'sku'
+      end
+
+      def by_sku(scope)
+        return scope unless sku?
+
+        scope.joins(:master).
+          where(Spree::Variant.table_name.to_s => { is_master: true }).
+          order("#{Spree::Variant.table_name}.sku #{order_direction}")
```
Comment (damianlegawiec) [inline core/app/sorters/spree/products/sort.rb:44]:
```suggestion
          order(Spree::Variant.table_name => { sku: order_direction })
```
this should work

Comment (szymoniwacz) [reply core/app/sorters/spree/products/sort.rb:44]: @damianlegawiec Tried that and it doesn't work.

File: api/spec/requests/spree/api/v2/storefront/products_spec.rb [hunk 2]
```diff
@@ -401,6 +403,40 @@
...
+
+        context 'descending order' do
+          before { get '/api/v2/storefront/products?sort=-sku' }
+
+          it_behaves_like 'returns 200 HTTP status'
+
+          it 'returns products sorted by master variant sku with descending order' do
+            expect(skus_array).to eq(skus_array.sort.reverse)
+            expect(json_response['data'].count).to      eq store.products.available.count
+            expect(json_response['data'].pluck(:id)).to eq store.products.available.joins(:master).
```
Comment (damianlegawiec) [inline api/spec/requests/spree/api/v2/storefront/products_spec.rb:433]: @szymoniwacz let's not use Arel queries in tests, we need to type down the ids by hand to perfectly sure :)

File: api/spec/requests/spree/api/v2/storefront/products_spec.rb [hunk 3]
```diff
@@ -451,6 +451,35 @@
           end
         end
       end
+
+      context 'sorting by sku' do
+        let(:skus_array) { json_response['data'].map { |p| p['attributes']['sku']} }
+        let(:available_products_with_master) { products + [product_with_option, in_stock_product, not_backorderable_product, product_with_property] }
+
+        context 'ascending order' do
+          before { get '/api/v2/storefront/products?sort=sku' }
```
Comment (damianlegawiec) [inline api/spec/requests/spree/api/v2/storefront/products_spec.rb:460]: @szymoniwacz let's also add stest 

1. where we include variants
2. where we sort by SKU and another param like updated at

Comment (szymoniwacz) [reply api/spec/requests/spree/api/v2/storefront/products_spec.rb:460]: @damianlegawiec Spree::BaseSorter is designed to sort only by one param.

**Review by damianlegawiec — APPROVED** [review]

**Pull request merged**